script/on_tick.lua

```diff
@@ -84,65 +94,107 @@ local function load_carriage(prop)
 	return NewCarriage
 end
 
-local function on_tick(event)
-	index = 1
-	for unit,prop in pairs(global.TrainsInTunnel) do
-
-		if prop.escape_done == true and prop.head_escaped == true then
-			prop.guideCar.destroy()
-			prop.TempTrain.destroy()
-			table.remove(global.TrainsInTunnel,index)
-			goto continue
-		end
+local function search_tunnel(train)
+	target = train.ghostCar.surface.find_entity("TrainTunnelT2-mask", train.ghostCar.position)
```

Comment:
So now we check each train's coordinates and see if we have anything over flying ghost car
To simplify we can set a timer(caculated by distance between tunnels / speed of train entering the tunnel) before flying a ghost car and ignore any checking, and let all the arriving logic at the end of timer

Reply:
need to know what speed refers to(ex. does 0.7 mean 0.7 block/s?)

Reply:
variable tick is ready for this already